flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/operators/sink/committables/CommittableCollectorSerializerTest.java

```diff
@@ -72,38 +73,86 @@ void testCommittableCollectorV1SerDe() throws IOException {
 
     @Test
     void testCommittableCollectorV2SerDe() throws IOException {
-        final CommittableCollector<Integer> committableCollector = new CommittableCollector<>(2, 3);
-        committableCollector.addMessage(new CommittableSummary<>(2, 3, 1L, 1, 1, 0));
-        committableCollector.addMessage(new CommittableSummary<>(2, 3, 2L, 1, 1, 0));
-        committableCollector.addMessage(new CommittableWithLineage<>(1, 1L, 2));
-        committableCollector.addMessage(new CommittableWithLineage<>(2, 2L, 2));
+        int subtaskId = 2;
```

Comment:
I think it would be better to add/adapt a test in the `CommitterOperatorTest`. You already have to expose more internals i.e. `@VisibleForTesting` in the CheckpointCommittableManagerImpl` which is a somewhat bad sign. 

In the end, we want to have a test where, a `CommitterOperater` with a subtaskId != 0 recovers the committables and sends them forward and we want to assert that the subtaskId is the one from the new `CommitterOperator`

Reply:
thanks, 
with [getSummary](https://github.com/apache/flink/pull/20979/files#r991004178) i managed to not expose new information from `CheckpointCommittableManagerImpl`. 

I do think that `CommittableCollectorSerializerTest` is a good place for this check (I;m asserting additional values only)
since the original issue FLINK-29509 was caused by indirectly by deserialization - now we check if we use corect constructor hence we have proper values. 

Additionally I will take a look at `CommitterOperatorTest`

Reply:
Hi @fapaul 
I have modify CommitterOperatorTest::testStateRestore test to make sure we use proper `subtaskId` after operator recovery.

Reply:
Ok sounds good to me 👍